scripts/generate-variables.mjs

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env node
+
+// TODO:
+// capture light/dark vars into their own fields
+import Parser from 'tree-sitter';
+import CSS from 'tree-sitter-css';
+import fs from 'fs';
+import process from 'process';
+
+process.stdin.on('data', (data) => {
+  parseCssToVariables(data.toString());
+});
+
+function parseCssToVariables(cssCode) {
+  const parser = new Parser();
+  parser.setLanguage(CSS);
+
+  const VARIABLE_SELECTOR = '.str-chat';
+  const LIGHT_THEME_SELECTOR = `.str-chat,
+.str-chat__theme-light`;
+  const DARK_THEME_SELECTOR = `.str-chat__theme-dark`;
+  const commentParser = new RegExp('/\\*\\s*(.+?)(?:\\. )?(Used in.+)?\\s*\\*/');
+  const colorVariableMatch = new RegExp(/green|blue|red|grey|yellow/);
+
+  const unrecognizedCssVariables = [];
+  const processedCssVariables = {
+    colors: {},
+    spacing: [],
+    borderRadius: [],
+    font: [],
+    components: {},
+    lightTheme: [],
+    darkTheme: [],
+  };
+
+  const tree = parser.parse(cssCode);
+
+  tree.rootNode.children.forEach((child) => {
+    if (child.type === 'rule_set') {
+      const selector = child.child(0).text;
+      if (
+        selector == VARIABLE_SELECTOR ||
+        selector == LIGHT_THEME_SELECTOR ||
+        selector === DARK_THEME_SELECTOR
+      ) {
+        const contents = child.child(1).children.slice(1, -1);
+
+        contents.forEach((child, index) => {
+          if (child.type === 'declaration') {
+            if (child.child(0).text.startsWith('--')) {
+              let comment = { description: '', usage: '' };
+              if (index > 0) {
+                let prevNode = contents[index - 1];
+                if (prevNode.type === 'comment') {
+                  const matches = prevNode.text.match(commentParser);
+                  comment.description = matches[1];
+                  comment.usage = matches[2];
+                }
+              }
+
+              const varName = child.child(0).text;
+
+              const variableInfo = {
+                varName,
+                varValue: child.children
+                  .slice(2, -1)
+                  .map((value) => value.text)
+                  .join(' '),
+              };
+              if (comment.description) {
+                variableInfo.description = comment.description;
+              }
+              if (comment.usage) {
+                variableInfo.usage = comment.usage;
+              }
+
+              if (selector === LIGHT_THEME_SELECTOR) {
+                processedCssVariables.lightTheme.push(variableInfo);
+                return;
+              }
+
+              if (selector === DARK_THEME_SELECTOR) {
```

Comment:
We sometimes use the dark theme selector inside components: https://github.com/GetStream/stream-chat-css/blob/theming2/src/v2/styles/MessageActionsBox/MessageActionsBox-theme.scss#L28 - I think that would categorize the `--str-chat__message-actions-box-box-shadow` with the theme variables

Reply:
But that's an issue with the CSS code, we probably shouldn't have the dark selector inside a component file, maybe we should create a theme variable for box shadow?

Reply:
Done